src/main/scala/com/vngrs/etl/Pipeline.scala

```diff
@@ -0,0 +1,94 @@
+package com.vngrs.etl
+
+import com.vngrs.etl.combiners.{Union, Zip}
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * Represents ETL Pipeline.
+  *
+  * @param rdd The [[org.apache.spark.rdd.RDD]] which will be ETLed
+  * @tparam A Type of the [[org.apache.spark.rdd.RDD]]
+  */
+final class Pipeline[A: ClassTag](val rdd: RDD[A]) {
```

Comment:
Yes, since we are exposing `rdd`, your suggestion is doable. However, there is one minor point, `Pipeline` takes `SparkContext` implicitly and gives to `Extract` explicitly so user has a nice DSL like just making spark context implicit val. However, when we make `Extract` to return `Pipeline` then user must call `apply` method and give spark context explicitly. Or we can make an implicit class which will handle that.

BTW, we can make whole design lazy so that `Extract` will not need a spark context immediately but again we simply don't have time to change design (otherwise we can not continue this project since we are stuck at this point for a long time) as we spoke. At least I don't have as you know :).